packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-module.ts

```diff
@@ -15,10 +15,37 @@
 // *****************************************************************************
 
 import { ContainerModule } from '@theia/core/shared/inversify';
-import { FileDialogService } from '../../browser/file-dialog/file-dialog-service';
+import { DefaultFileDialogService, FileDialogService } from '../../browser/file-dialog/file-dialog-service';
 import { ElectronFileDialogService } from './electron-file-dialog-service';
+import { LocationListRenderer, LocationListRendererFactory, LocationListRendererOptions } from '../../browser/location';
+import { FileDialogHiddenFilesToggleRenderer, HiddenFilesToggleRendererFactory } from '../../browser/file-dialog/file-dialog-hidden-files-renderer';
+import { FileDialogTree } from '../../browser/file-dialog/file-dialog-tree';
+import { FileDialogTreeFiltersRenderer, FileDialogTreeFiltersRendererFactory, FileDialogTreeFiltersRendererOptions } from '../../browser/file-dialog';
 
 export default new ContainerModule(bind => {
-    bind(ElectronFileDialogService).toSelf().inSingletonScope();
-    bind(FileDialogService).toService(ElectronFileDialogService);
+    if (window.electronTheiaFilesystem.useNativeDialogs) {
+        bind(ElectronFileDialogService).toSelf().inSingletonScope();
+        bind(FileDialogService).toService(ElectronFileDialogService);
+    } else {
+        bind(DefaultFileDialogService).toSelf().inSingletonScope();
```

Comment:
This loos like it would be a copy of the browser case. If that is true, can we share this code in a function somewhere?

Reply:
I moved the logic of whether native elements are used away from the module into the implementation. We now call the superclass of `ElectronFileDialogService` if native elements should not be used. This should also make it easier to move away from the env variable approach in future work.